compute_sdk/globus_compute_sdk/serialize/base.py

```diff
@@ -17,28 +18,59 @@ def __init_subclass__(cls):
         if len(cls.identifier) != IDENTIFIER_LENGTH:
             raise ValueError(f"Identifiers must be {IDENTIFIER_LENGTH} characters long")
 
+    #: The unique, 3-character string that identifies this strategy. The first two
+    #: characters are the effective identifier; the third must always be the newline
+    #: character ``"\n"``.
     identifier: t.ClassVar[str]
+    #: If ``True``, this strategy works with callable Python objects - functions,
+    #: classes that implement ``__call__``, etc. If ``False``, this strategy works with
+    #: all other Python objects (ie, arguments and keyword arguments in submissions).
+    #: Note that a single ``SerializationStrategy`` cannot support both callables and
+    #: non-callables.
     for_code: t.ClassVar[bool]
 
+    @abstractmethod
+    def serialize(self, data: t.Any) -> str:
+        """
+        Converts arbitrary Python data to a string representation.
+
+        :param data: If ``for_code``, any callable Python object; if not
+            ``for_code``, any other Python object. Serializers are not
+            guaranteed to work with every callable or non-callable object.
+
+        :returns: The string representation, which can be deserialized again by an
+            instance of this strategy.
+
+        :raises: A ``SerializationStrategy`` may raise for various reasons if it
+            cannot handle a given Python object.
+        """
+        raise NotImplementedError("Concrete class did not implement serialize")
+
+    @abstractmethod
+    def deserialize(self, payload: str) -> t.Any:
+        """
+        Converts serialized string data back into the Python object it represents.
+        Inverse of ``serialize``.
+
+        :param payload: Serialized data generated by an instance of this strategy.
+
+        :returns: The Python object represented by the payload.
+
+        :raises: If the payload wasn't serialized by an instance of this strategy.
```

Comment:
```suggestion
        :raises: If the payload wasn't serialized by an instance of this strategy or encountering other unexpected errors.
```

Deserialization fails for version conflicts, ie. the typical '19 arguments expecting 21' one?